category/functor.lean

```diff
@@ -0,0 +1,120 @@
+/-
+Copyright (c) 2017 Simon Hudon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Simon Hudon
+
+Standard identity and composition functors
+-/
+import tactic.ext
+import category.basic
+
+universe variables u v w u' v' w'
+
+section functor
+
+variables {F : Type u → Type v}
+variables {α β γ : Type u}
+variables [functor F] [is_lawful_functor F]
+
+lemma functor.map_id : functor.map id = (id : F α → F α) :=
+by apply funext; apply id_map
+
+lemma functor.map_comp_map (f : α → β) (g : β → γ) :
+  (functor.map g ∘ functor.map f : F α → F γ) = functor.map (g ∘ f) :=
+by apply funext; intro; rw comp_map
+
+@[norm]
```

Comment:
this is `map_map` and in `functor_norm`

Reply:
What's the reason for having `functor_norm` be distinct from `norm`?

Reply:
Also, can we put `map_map` in the `functor` namespace? It clashes with names in `multiset`